test/arbitrum/JPEGzMainnetOracle.t.sol

```diff
@@ -0,0 +1,505 @@
+      liquidationPenalty,
+      address(jpegzOracle),
+      jpegz,
+      address(weth),
+      address(ethOracle),
+      address(ethOracle),
+      treasury,
+      0 ether
+    );
+
+    orchestrator.addTCAPVault(jpegz, ethVault);
+  }
+
+  function testConstructor_ShouldSetParams_WhenInitialized() public {
+    assertEq(address(orchestrator), ethVault.owner());
+    assertEq(divisor, ethVault.divisor());
+    assertEq(ratio, ethVault.ratio());
+    assertEq(burnFee, ethVault.burnFee());
+    assertEq(burnFee, ethVault.mintFee());
+    assertEq(liquidationPenalty, ethVault.liquidationPenalty());
+    assertEq(address(jpegzOracle), address(ethVault.tcapOracle()));
+    assertEq(address(jpegz), address(ethVault.TCAPToken()));
+    assertEq(address(weth), address(ethVault.collateralContract()));
+    assertEq(address(ethOracle), address(ethVault.collateralPriceOracle()));
+    assertEq(address(ethOracle), address(ethVault.ETHPriceOracle()));
+    assertEq(treasury, ethVault.treasury());
+    assertEq(0 ether, ethVault.minimumTCAP());
+  }
+
+  function testConstructor_ShouldRevert_WhenBurnFeeIsHigh(uint256 _burnFee)
+    public
+  {
+    if (_burnFee > 1000) {
+      vm.expectRevert("VaultHandler::constructor: fee higher than MAX_FEE");
+    }
+
+    ETHVaultHandler vault = new ETHVaultHandler(
+      orchestrator,
+      divisor,
+      ratio,
+      _burnFee,
+      mintFee,
+      liquidationPenalty,
+      address(jpegzOracle),
+      jpegz,
+      address(weth),
+      address(ethOracle),
+      address(ethOracle),
+      treasury,
+      0
+    );
+
+    if (!(_burnFee > 1000)) {
+      assertEq(_burnFee, vault.burnFee());
+    }
+  }
+
+  function testConstructor_ShouldRevert_WhenMintFeeIsHigh(uint256 _mintFee)
+    public
+  {
+    vm.assume(_mintFee > 1000);
+
+    vm.expectRevert("VaultHandler::constructor: fee higher than MAX_FEE");
+    new ETHVaultHandler(
+      orchestrator,
+      divisor,
+      ratio,
+      burnFee,
+      _mintFee,
+      liquidationPenalty,
+      address(jpegzOracle),
+      jpegz,
+      address(weth),
+      address(ethOracle),
+      address(ethOracle),
+      treasury,
+      0
+    );
+  }
+
+  function testConstructor_ShouldRevert_WhenLiquidationPenaltyIsHigh(
+    uint256 _liquidationPenalty,
+    uint256 _ratio
+  ) public {
+    if (_liquidationPenalty + 100 < _liquidationPenalty) {
+      return;
+    }
+
+    if ((_liquidationPenalty + 100) >= _ratio) {
+      vm.expectRevert(
+        "VaultHandler::constructor: liquidation penalty too high"
+      );
+    }
+
+    ETHVaultHandler vault = new ETHVaultHandler(
+      orchestrator,
+      divisor,
+      _ratio,
+      burnFee,
+      mintFee,
+      _liquidationPenalty,
+      address(jpegzOracle),
+      jpegz,
+      address(weth),
+      address(ethOracle),
+      address(ethOracle),
+      treasury,
+      0
+    );
+
+    if (!((_liquidationPenalty + 100) >= _ratio)) {
+      assertEq(_liquidationPenalty, vault.liquidationPenalty());
```

Comment:
Please split this as a separate as this test is different from its name i.e `ShouldRevert_WhenLiquidationPenaltyIsHigh`.